lib/src/http/common/request.dart

```diff
@@ -769,7 +770,7 @@ abstract class CommonRequest extends Object
       // Enforce a timeout threshold if set.
       Timer timeout;
       if (timeoutThreshold != null) {
-        timeout = Timer(timeoutThreshold, _timeoutRequest);
+        timeout = Timer(autoRetry.timeoutThreshold, _timeoutRequest);
```

Comment:
Feels a little awkward to me, but unit tests should protect from mistakes using the request's timeoutThreshold.